### PR TITLE
CAMEL-17031 - Performances/Profiling: Avoid boolean allocation in Eve…

### PR DESCRIPTION
**Pull request by oscerd**

### CODE REVIEW

File: core/camel-support/src/main/java/org/apache/camel/support/EventHelper.java [hunk 1]
```diff
@@ -44,6 +44,7 @@
     // be left as-is.
 
     private static final Logger LOG = LoggerFactory.getLogger(EventHelper.class);
+    private static volatile boolean started = true;
```
Comment (davsclaus) [inline core/camel-support/src/main/java/org/apache/camel/support/EventHelper.java:47]: should be false by default

File: core/camel-support/src/main/java/org/apache/camel/support/EventHelper.java [hunk 2]
```diff
@@ -1300,7 +1301,7 @@ public static boolean notifyStepFailed(CamelContext context, Exchange exchange,
 
     private static boolean doNotifyEvent(EventNotifier notifier, CamelEvent event) {
         // only notify if notifier is started
-        boolean started = true;
+        started = true;
         if (notifier instanceof StatefulService) {
```
Comment (davsclaus) [inline core/camel-support/src/main/java/org/apache/camel/support/EventHelper.java:1305]: if (!started && notifier ....)

File: core/camel-support/src/main/java/org/apache/camel/support/EventHelper.java [hunk 3]
```diff
@@ -1300,7 +1301,7 @@ public static boolean notifyStepFailed(CamelContext context, Exchange exchange,
 
     private static boolean doNotifyEvent(EventNotifier notifier, CamelEvent event) {
         // only notify if notifier is started
-        boolean started = true;
+        started = true;
```
Comment (davsclaus) [inline core/camel-support/src/main/java/org/apache/camel/support/EventHelper.java:1304]: remove started = true as we need to check the notifier instance.

And if its NOT a stateful service then we can regard it as started

**Review by davsclaus — CHANGES_REQUESTED** [review]

**Comment by oscerd** [comment]
See #6194

**Pull request closed without merging**